Add health endpoint API support

awendelin-work opened the pull request:
Add a fetch_health method to TspClient to allow a user to fetch the health status of the trace server as described in the TSP.

I have issues when running the tests locally, but I do not think they are related to the changes here.
I get these failures when running pytest:
```
FAILED test_tsp.py::TestTspClient::test_fetch_configurations_none - assert []                                                                                              
FAILED test_tsp.py::TestTspClient::test_posted_configuration_deleted - assert []
```

I am running the trace-server in a docker container built from the incubator commit ae0a7d93a6a2d6fa339f35ccd430e25dce1961c5 with only this repository mounted from the host, so it is possible that I am breaking some assumption.

Resolves #66

MatthewKhouzam reviewed (approved):
lgtm

bhufmann commented:
> Add a fetch_health method to TspClient to allow a user to fetch the health status of the trace server as described in the TSP.
> 
> I have issues when running the tests locally, but I do not think they are related to the changes here. I get these failures when running pytest:
> 
> ```
> FAILED test_tsp.py::TestTspClient::test_fetch_configurations_none - assert []                                                                                              
> FAILED test_tsp.py::TestTspClient::test_posted_configuration_deleted - assert []
> ```
> 
> I am running the trace-server in a docker container built from the incubator commit ae0a7d93a6a2d6fa339f35ccd430e25dce1961c5 with only this repository mounted from the host, so it is possible that I am breaking some assumption.
> 
> Resolves #66

I don't see any errors when running the tests using a local trace server. So, it might be related to your setup. No, worries for this PR.

bhufmann reviewed (changes requested):
Code looks good to me. Please update the usage section of the README for this:
https://github.com/eclipse-cdt-cloud/tsp-python-client?tab=readme-ov-file#usage

awendelin-work commented:
> Code looks good to me. Please update the usage section of the README for this: https://github.com/eclipse-cdt-cloud/tsp-python-client?tab=readme-ov-file#usage

I did not include a CLI update in the previous version, I merely updated the Python API, so nothing changed from CLI point of view.

I take it that you want a CLI option for health. I have added a `--get-health` option to the CLI  and updated the README with it.


> > I have issues when running the tests locally, but I do not think they are related to the changes here. I get these failures when running pytest:
> I don't see any errors when running the tests using a local trace server. So, it might be related to your setup. No, worries for this PR.

To me it makes sense that they fail from looking at the code.

`test_tsp.py::TestTspClient::test_fetch_configurations_none` expects no configurations to be available in the trace-server. The trace-server responds with a 200 and an empty list of configurations.

The test then asserts on the list of returned configurations `assert response.model.configuration_set` which is an empty list in this case. An empty list is evaluated to false in Python, so the assertion fails.

```
> /home/<redacted>/builds/tsp-python-client/test_tsp.py(426)test_fetch_configurations_none()
-> assert response.model.configuration_set
(Pdb) p response.model
<tsp.configuration_set.ConfigurationSet object at 0x7feee1adc490>
(Pdb) p response.model.configuration_set
[]
(Pdb) p [] == True
False
```

So to me it looks like the assertion should really be `assert response.model.configuration_set == []` to be correct. The current assert checks for the presence of configurations rather than the absence of.

The same pattern can be seen in `test_tsp.py::TestTspClient::test_posted_configuration_deleted` where a configuration is posted successfully to the server and subsequently deleted successfully. The test then fetches all available configurations and does the same assert `assert response.model.configuration_set` which fails because there are no configurations in the list (but that is what is actually expected).

bhufmann commented:
> > Code looks good to me. Please update the usage section of the README for this: https://github.com/eclipse-cdt-cloud/tsp-python-client?tab=readme-ov-file#usage
> 
> I did not include a CLI update in the previous version, I merely updated the Python API, so nothing changed from CLI point of view.
> 
> I take it that you want a CLI option for health. I have added a `--get-health` option to the CLI and updated the README with it.
> 
> > > I have issues when running the tests locally, but I do not think they are related to the changes here. I get these failures when running pytest:
> > > I don't see any errors when running the tests using a local trace server. So, it might be related to your setup. No, worries for this PR.
> 
> To me it makes sense that they fail from looking at the code.
> 
> `test_tsp.py::TestTspClient::test_fetch_configurations_none` expects no configurations to be available in the trace-server. The trace-server responds with a 200 and an empty list of configurations.
> 
> The test then asserts on the list of returned configurations `assert response.model.configuration_set` which is an empty list in this case. An empty list is evaluated to false in Python, so the assertion fails.
> 
> ```
> > /home/<redacted>/builds/tsp-python-client/test_tsp.py(426)test_fetch_configurations_none()
> -> assert response.model.configuration_set
> (Pdb) p response.model
> <tsp.configuration_set.ConfigurationSet object at 0x7feee1adc490>
> (Pdb) p response.model.configuration_set
> []
> (Pdb) p [] == True
> False
> ```
> 
> So to me it looks like the assertion should really be `assert response.model.configuration_set == []` to be correct. The current assert checks for the presence of configurations rather than the absence of.
> 
> The same pattern can be seen in `test_tsp.py::TestTspClient::test_posted_configuration_deleted` where a configuration is posted successfully to the server and subsequently deleted successfully. The test then fetches all available configurations and does the same assert `assert response.model.configuration_set` which fails because there are no configurations in the list (but that is what is actually expected).

That makes sense. I noticed that I have xml analysis loaded in my server workspace and hence it worked. After removing them get the same error. The tests need to start from a clean server workspace and isolated server. It already deletes traces/experiment at the beginning of the tests. But this is not ideal because if the same server workspace is used for regular work. 

Anyways, I'll upload fix for the tests as you proposed. It works with and without pre-loaded xml anlaysis.

bhufmann reviewed (approved):
Looks good to me. Thanks for the contribution.

MatthewKhouzam reviewed (approved):
Looks good here, thanks for updating the doc!
